### DISCUSSION TITLE
Better control over the list of clients

### DISCUSSION THREAD

**Discussion by mrchimp** [scalar/scalar#807] [Ideas]
The wide range of clients available is great but is a little problematic as listing them might imply some sort of support. In our use case we have SDKs in a couple of languages and would like to only provide examples in those same languages to avoid confusion (even though the examples wouldn't use our SDKs and the API would work in any language). 

The updates for #707 are a welcome addition and this makes sense if there a few clients that you specifically want to exclude. However if as we only want to display a specific few clients, we need to maintain a large list of clients that we're *not* using.

Also because some of the clients use the same key (`curl` and `native` for example) the current functionality of the `hiddenClients` param doesn't give complete control.

It would also be good to be able to control the list of clients that are displayed as icons. I see in the code that this is defined as two lists but this might be simpler to define as one list and dynamically remove the later elements for display on smaller screens. That way we could provide a single array in the configuration.

**Comment by hanspagel** [comment]
I think we should do this. 👍

**Comment by linhub15** [comment]
Although verbose, it seems like we can already show the exact clients what we want.
Here's an example with `@scalar/hono-api-reference`

```js
apiReference({
  hiddenClients: {
    c: true, // hides all for c
    javascript: ["xhr", "axios", "jquery"], // hide select clients, `fetch` is allowed
    shell: [] // show all for shell
    node: undefined // show all for node
  }
})
```

**Comment by amritk** [comment]
taggin in @hwkr

**Reply by hwkr** [reply]
Yes! I think we're already doing a bunch of this, better configuration of the featured clients we should make a ticket for.
